packages/atomic/src/components/common/facets/numeric-facet/value-link.spec.ts

```diff
@@ -0,0 +1,301 @@
+import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
+import {createTestI18n} from '@/vitest-utils/testing-helpers/i18n-utils';
+import '@vitest/browser/matchers.d.ts';
+import {html} from 'lit';
+import {vi, expect, describe, beforeAll, beforeEach, it} from 'vitest';
+import {renderFacetValueCheckbox} from '../facet-value-checkbox/facet-value-checkbox';
+import {renderFacetValueLabelHighlight} from '../facet-value-label-highlight/facet-value-label-highlight';
+import {renderFacetValueLink} from '../facet-value-link/facet-value-link';
+import {formatHumanReadable} from './formatter';
+import {
+  renderNumericFacetValue,
+  NumericFacetValueLinkProps,
+} from './value-link';
+
+// Mock the dependencies
+vi.mock('../facet-value-checkbox/facet-value-checkbox', {spy: true});
+vi.mock('../facet-value-link/facet-value-link', {spy: true});
+vi.mock('../facet-value-label-highlight/facet-value-label-highlight', {
+  spy: true,
+});
+vi.mock('./formatter', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('./formatter')>();
+  return {
+    ...actual,
+    formatHumanReadable: vi.fn(),
+  };
+});
+
+describe('#renderNumericFacetValue', () => {
+  let i18n: Awaited<ReturnType<typeof createTestI18n>>;
+  let mockOnClick: ReturnType<typeof vi.fn>;
+  let mockLogger: Pick<Console, 'error'>;
+  let mockFormatter: NumberFormatter;
+  let defaultProps: NumericFacetValueLinkProps;
+
+  beforeAll(async () => {
+    i18n = await createTestI18n();
+  });
+
+  beforeEach(() => {
+    mockOnClick = vi.fn();
+    mockLogger = {error: vi.fn()};
+    mockFormatter = vi.fn().mockReturnValue('formatted number');
+
+    defaultProps = {
+      field: 'price',
+      facetValue: {
+        start: 10,
+        end: 20,
+        endInclusive: true,
+        state: 'idle',
+        numberOfResults: 42,
+      },
+      manualRanges: [],
+      i18n,
+      logger: mockLogger,
+      formatter: mockFormatter,
+      onClick: mockOnClick,
+      displayValuesAs: 'checkbox',
+    };
+
+    vi.clearAllMocks();
+    vi.mocked(formatHumanReadable).mockReturnValue('10 to 20');
+  });
+
+  const renderComponent = async (
+    props: Partial<NumericFacetValueLinkProps> = {}
+  ) => {
+    const mergedProps = {...defaultProps, ...props};
+    return await renderFunctionFixture(
+      html`${renderNumericFacetValue({props: mergedProps})}`
+    );
+  };
+
+  describe('when displayValuesAs is "checkbox"', () => {
+    beforeEach(() => {
+      // Mock the checkbox render function to return a recognizable template
+      vi.mocked(renderFacetValueCheckbox).mockReturnValue(
+        () => html`<div data-testid="checkbox-container">checkbox content</div>`
+      );
+      vi.mocked(renderFacetValueLabelHighlight).mockReturnValue(
+        html`<span data-testid="label-highlight">highlighted label</span>`
+      );
+    });
+
+    it('should call formatHumanReadable with correct props', async () => {
```

Comment:
This will be done regardless of the displayValuesAs value, so it should probably be outside of this describe block.